akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -40,8 +43,41 @@ object ActorTestKit {
   def apply(): ActorTestKit =
     new ActorTestKit(
       name = TestKitUtils.testNameFromCallStack(classOf[ActorTestKit]),
-      config = ApplicationTestConfig,
+      ActorSystemSetup.create(BootstrapSetup(ApplicationTestConfig)),
+      settings = None)
+
+  /**
+   * Create a testkit named based on the provided ActorSystem.
+   *
+   * It will create an [[akka.actor.typed.ActorSystem]] with this name,
+   * e.g. threads will include the name, and the related actor sustem setup.
+   * When the test has completed you should terminate the `ActorSystem` and
+   * the testkit with [[ActorTestKit#shutdownTestKit]].
+   *
+   * Config loaded from `system.settings.config` if that exists, otherwise
+   * using default configuration from the reference.conf resources that ship with the Akka libraries.
+   */
+  def apply(system: ActorSystem[_]): ActorTestKit = {
```

Comment:
This is only using the name of that `system`, which I don't see the purpose of.
We also want to add support for passing in a custom ActorSystem and use that instead of creating an ActorSystem. That means that the testkit still needs to spawn ActorTestKitGuardian in that system.
So when passing in a system it should
```
val testKitGuardian: ActorRef[ActorTestKitGuardian.TestKitCommand] =
  system.systemActorOf(ActorTestKitGuardian.testKitGuardian, name = "test")
```

The nice thing is that for the other option when the testkit is creating the ActorSystem that is also an `ActorRef[ActorTestKitGuardian.TestKitCommand]`

```
val testKitGuardian: ActorRef[ActorTestKitGuardian.TestKitCommand] =
  ActorSystem(ActorTestKitGuardian.testKitGuardian, name, config)
```